zebra-state/src/in_memory/chains.rs

```diff
@@ -3,18 +3,97 @@ use crate::{Request, Response};
 use futures::prelude::*;
 use std::{
     pin::Pin,
+    sync::Arc,
     task::{Context, Poll},
 };
-use tower::Service;
+use tower::{Service, ServiceExt};
+use zebra_chain::{
+    block::{Block, BlockHeaderHash},
+    types::BlockHeight,
+};
+
+/// A representation of the chain state at a given height
+#[derive(Clone, Debug)]
+struct ChainState {
+    block: Arc<Block>,
+}
```

Comment:
For the sake of other reviewers:
```suggestion
    block: Arc<Block>,
    // TODO: add more state here
}
```